### DISCUSSION TITLE
Suggest developing a behavior tree editor

### DISCUSSION THREAD

**Discussion by MadFlyFish** [kagenash1/godot-behavior-tree#8] [Ideas]
**Thanks for this great addon!**

-  I have being using it for a couple of weeks, and still looking forward to more  future updates. Recently I am developing a tool, which uses Xmind and excel as an tree editor and create behavior tree dynamically in run-time, mostly based on your addon‘s code.

- That helps my work a lot. Therefore I am introducing your addons to my Godot developer friends in China, and they show great interest too.

- But still, an external editor is not so handy as desired. I suggest developing a behavior tree editor, which makes use of Godot's internal editor . That may make AI editing  and debugging a lot easier.

- I am look forward to some progress on this. LOL

Here are some screenshots of my external editor. I am grad to share this tool if someone needs.

![image](https://user-images.githubusercontent.com/36298330/111722049-0a0f9080-889c-11eb-865e-ea2bdc47456e.png)

![image](https://user-images.githubusercontent.com/36298330/111722079-185dac80-889c-11eb-96f1-5506ccdb0395.png)

![image](https://user-images.githubusercontent.com/36298330/111722099-21e71480-889c-11eb-84cc-3551c5e0d12f.png)

![image](https://user-images.githubusercontent.com/36298330/111722447-d5500900-889c-11eb-9e59-577e3ab2c591.png)

![image](https://user-images.githubusercontent.com/36298330/111722489-eac53300-889c-11eb-8b9b-3f6c0cd0019f.png)

![image](https://user-images.githubusercontent.com/36298330/111722509-f153aa80-889c-11eb-9797-d59abd682084.png)

**Comment by kagenash1** [comment]
Great, this is part of the plan for 1.2.0.
It seems you have made some progress in terms of editor, maybe your system can be adapted to some internal editor? I would be happy to merge it if it works well.

**Reply by MadFlyFish** [reply]
> Great, this is part of the plan for 1.2.0.
> It seems you have made some progress in terms of editor, maybe your system can be adapted to some internal editor? I would be happy to merge it if it works well.

- yep, I am considering  working on that. But still need some more study and assistance. I start using Godot in spare time 3 months ago, and desire for a handy behavior tree solution. 

- So far your addon is the simplest one that I found in asset store. There is still not a AI solution with great usability, so I really hope this addon will keep updated.

**Comment by kagenash1** [comment]
I created a new issue which addresses this. If you want to, you can assign yourself :)
